src/arkode/arkode.c

```diff
@@ -1485,21 +1484,6 @@ ARKodeMem arkCreate(SUNContext sunctx)
   ark_mem->lrw += ARK_ADAPT_LRW;
   ark_mem->liw += ARK_ADAPT_LIW;
 
-  /* Allocate default step controller (PID) and note storage */
-  ark_mem->hadapt_mem->hcontroller = SUNAdaptController_PID(sunctx);
```

Comment:
Are there any `Set` functions that could be called after `Create` that assume a controller exists?

Reply:
I see, this a little different from what's done with the interpolation modules where attaching the default is done in `arkInitialSetup`. In this case, `SetDefaults` will (always?) create a controller so one will exist after `Create` is called.

Reply:
Almost.

It is the responsibility of a time-stepper module's `SetDefaults` routine to allocate a controller if time adaptivity should be enabled.  Steppers that do not support adaptivity can leave it `NULL` (e.g., SPRKStep), and ARKODE will now refrain from calling the [`NULL`-valued] controller.

Reply:
Sounds good. I think there are some `Set` functions that need to be updated to also check for a `NULL` controller e.g., `ARKodeSetInitStep`, `ARKodeSetErrorBias`, etc.

Reply:
Good catch.  I just updated the PR to add guards to protect against `NULL`-valued SUNAdaptController objects.